src/charm.py

```diff
@@ -71,22 +76,23 @@ def _on_config_changed(self, _):
         for key in config:
             self._stored.config[key] = config[key]
 
-        try:
-            if config.get("ssl_cert"):
-                cert_path = os.path.join("/etc/nginx/ssl", "server.crt")
-                cert = b64decode(config["ssl_cert"])
-                write_file(cert_path, cert, 0o644)
-
-            if config.get("ssl_key"):
-                key_path = os.path.join("/etc/nginx/ssl", "server.key")
-                key = b64decode(config["ssl_key"])
-                write_file(key_path, key, 0o640)
-
-            if config.get("ssl_ca"):
-                ca_cert = config["ssl_ca"]
-                install_ca_cert(b64decode(ca_cert))
-        except CAInstallError as e:
-            logger.error("CA installation error: %s", e)
+        ssl_cert_path = "/etc/nginx/ssl/server.crt"
+        if config.get("ssl_cert"):
+            atomic_write_root_file(ssl_cert_path, b64decode(config["ssl_cert"]), 0o644)
+        else:
+            if os.path.exists(ssl_cert_path):
+                os.remove(ssl_cert_path)
```

Comment:
I think the general recommendation for removing files is to try removing it and ignore file not found errors.  Otherwise the file could be deleted by something else between checking it exists and removing it.  Low priority here, but good practice.  For example:

```suggestion
            try:
                os.remove(ssl_cert_path)
            except FileNotFoundError:
                pass
```